src/serac/numerics/functional/isotropic_tensor.hpp

```diff
@@ -0,0 +1,457 @@
+SERAC_HOST_DEVICE constexpr auto ddot(const isotropic_tensor<S, m, m>& I, const tensor<T, m, m>& A)
+{
+  return I.value * tr(A);
+}
+
+/**
+ * @brief return the symmetric part of an isotropic tensor
+ *
+ * @tparam T the types stored in the isotropic tensor
+ * @tparam m the number of rows and columns in I
+ * @param I the isotropic tensor to symmetrize
+ * @return a copy of I (isotropic matrices are already symmetric)
+ */
+template <typename T, int m>
+SERAC_HOST_DEVICE constexpr auto sym(const isotropic_tensor<T, m, m>& I)
+{
+  return I;
+}
+
+/**
+ * @brief return the antisymmetric part of an isotropic tensor
+ *
+ * @tparam T the types stored in the isotropic tensor
+ * @tparam m the number of rows and columns in I
+ * @return `zero` (isotropic matrices are symmetric, so the antisymmetric part is identically zero)
+ */
+template <typename T, int m>
+SERAC_HOST_DEVICE constexpr auto antisym(const isotropic_tensor<T, m, m>&)
+{
+  return zero{};
+}
+
+/**
+ * @brief calculate the trace of an isotropic tensor
+ *
+ * @tparam T the types stored in the isotropic tensor
+ * @tparam m the number of rows and columns in I
+ * @param I the isotropic tensor to compute the trace of
+ * @return the sum of the diagonal entries of I
+ */
+template <typename T, int m>
+SERAC_HOST_DEVICE constexpr auto tr(const isotropic_tensor<T, m, m>& I)
+{
+  return I.value * m;
+}
+
+/**
+ * @brief return the transpose of an isotropic tensor
+ *
+ * @tparam T the types stored in the isotropic tensor
+ * @tparam m the number of rows and columns in I
+ * @param I the isotropic tensor to compute the trace of
+ * @return a copy of I (isotropic matrices are symmetric)
+ */
+template <typename T, int m>
+SERAC_HOST_DEVICE constexpr auto transpose(const isotropic_tensor<T, m, m>& I)
+{
+  return I;
+}
+
+/**
+ * @brief compute the determinant of an isotropic tensor
+ *
+ * @tparam T the types stored in the isotropic tensor
+ * @tparam m the number of rows and columns in I
+ * @param I the isotropic tensor to compute the determinant of
+ */
+template <typename T, int m>
+SERAC_HOST_DEVICE constexpr auto det(const isotropic_tensor<T, m, m>& I)
+{
+  return std::pow(I.value, m);
+}
+
+/**
+ * @brief compute the Frobenius norm (sqrt(tr(dot(transpose(I), I)))) of an isotropic tensor
+ *
+ * @tparam T the types stored in the isotropic tensor
+ * @tparam m the number of rows and columns in I
+ * @param I the isotropic tensor to compute the norm of
+ */
+template <typename T, int m>
+SERAC_HOST_DEVICE constexpr auto norm(const isotropic_tensor<T, m, m>& I)
+{
+  return sqrt(I.value * I.value * m);
+}
+
+/**
+ * @brief compute the squared Frobenius norm (tr(dot(transpose(I), I))) of an isotropic tensor
+ *
+ * @tparam T the types stored in the isotropic tensor
+ * @tparam m the number of rows and columns in I
+ * @param I the isotropic tensor to compute the squared norm of
+ */
+template <typename T, int m>
+SERAC_HOST_DEVICE constexpr auto sqnorm(const isotropic_tensor<T, m, m>& I)
```

Comment:
Is this common shorthand for this? If so, I'm fine with it. Otherwise, I think a more verbose name (`squared_norm`) is more appropriate.

Reply:
not that I know of, some of these things were just stolen from some old code of mine back in grad school. I think `squared_norm` is better (or do we want this function at all?)

Reply:
I'm OK either deleting it or renaming it.